clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -253,8 +254,8 @@ public void reenqueue(RecordBatch batch, long now) {
      * partition will be ready; Also return the flag for whether there are any unknown leaders for the accumulated
      * partition batches.
      * <p>
-     * A destination node is ready to send data if ANY one of its partition is not backing off the send and ANY of the
-     * following are true :
+     * A destination node is ready to send data if ANY one of its partition is not backing off the send and has no
+     * in flight batches and ANY of the following are true :
```

Comment:
The in-flight batches comment I think only applies when `sendInOrder` is true right? Also, it may be unclear to some what in-flight batch means.